lib/roda/plugins/exception_page.rb

```diff
@@ -84,10 +91,18 @@ module RodaPlugins
     # Used under the modified BSD license:
     # http://www.xfree86.org/3.3.6/COPYRIGHT2.html#5
     module ExceptionPage
-      def self.load_dependencies(app)
+      def self.load_dependencies(app, opts=OPTS)
         app.plugin :h
       end
 
+      # Use the CSS given as the :css option as the exception page's CSS and the
+      # JavaScript given as the :js option as the exception page's JavaScript,
+      # falling back to the defaults when not given.
+      def self.configure(app, opts=OPTS)
+        app.opts[:exception_page_css] = opts.fetch(:css, ExceptionPage.css)
```

Comment:
I don't think storing options for this is good.  If users want to override the CSS/JS, they can override the related class methods (or load a plugin that overrides them).

Reply:
Overriding the method was what I was trying to avoid. Currently, it is the only way to do this without breaking out to static files. I felt that introducing a seam through configuration made for a nice, clean way to give this first-class status in Roda.

Since many Roda apps can be combined together, I think there is value in allowing each app to contain it's own configuration for the plugin instead of relying on a global ~variable~ method.

If you disagree, that's okay. I will not be offended if you choose to decline this change.

That being said, when encountering the override that Bridgetown is doing, my initial reaction was, "wow, this could really use a blessed way of achieving this goal." Since plugins are configurable by design, this felt like an oversight.

Reply:
> Overriding the method was what I was trying to avoid. Currently, it is the only way to do this without breaking out to static files. I felt that introducing a seam through configuration made for a nice, clean way to give this first-class status in Roda.

I think it's unlikely most users will want to replace all of the CSS/JS.  Frameworks like Bridgetown might, but they are better off overriding the methods.  Prepending or appending to the CSS/JS is something more users may like to do, but that's not what the options in the pull request implement.

> Since many Roda apps can be combined together, I think there is value in allowing each app to contain it's own configuration for the plugin instead of relying on a global ~variable~ method.

Currently `ExceptionPage.{css,js}` is a global method, with the usual issues. Having Roda class methods for that allows for each app to customize the CSS/JS used for the page, so it no longer relies on a global method.

Both of us want the Roda class methods added.  You just also want plugin options to set the default.  However, I don't think it's a good idea, because you cannot easily augment the default behavior, as you saw in the problems updating the documentation for this. 

I could potentially consider other plugin options that didn't just specify the CSS or JS to use, but allowed you to modify the default (including full replacement). One possibility is a plugin option accepting a proc that is called with the default CSS/JS and returns the CSS/JS to use.

> Since plugins are configurable by design, this felt like an oversight.

It's not an oversight.  The exception_page plugin is based on Rack::ShowExceptions, which is similarly limited in terms of configuration options.  It is a developer-only facing tool, and this is the first request for such customization since the plugin was added over 4 years ago.  So I'm still of the opinion that there is limited need for such customization, and overriding the class methods is probably sufficient considering that.

Reply:
Alrighty! Let me think through your feedback and see what I come up with. Thank you for the detailed reply. 🙏